neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/embedded/impl/EmbeddedNeo4jGraphDatabaseFactory.java

```diff
@@ -86,4 +142,83 @@ private void setConfigurationFromLocation(GraphDatabaseBuilder builder, URL cfgL
 			builder.loadPropertiesFromURL( cfgLocation );
 		}
 	}
+
+	static void shutdownGraphDatabaseService(GraphDatabaseService neo4jDb) {
+		GraphDatabaseAPI sq = (GraphDatabaseAPI) neo4jDb;
+		String key = sq.getStoreDir().getAbsolutePath();
+		if ( GRAPH_DATABASE_SERVICE_MAP.containsKey( key ) ) {
+			synchronized (GRAPH_DATABASE_SERVICE_MAP) {
```

Comment:
Are you should you need to a nested synchronized?

Reply:
I try organize queue. You right. Thanks!